app/helpers/injection_helper.rb

```diff
@@ -51,6 +51,18 @@ def inject_enterprise_attributes
     render partial: "json/injection_ams", locals: {name: 'enterpriseAttributes', json: "#{@enterprise_attributes.to_json}"}
   end
 
+  def inject_orders_by_distributor
+    # Convert ActiveRecord::Relation to array for serialization
+    data_array = spree_current_user.orders_by_distributor.to_a
+    data_array.each do |enterprise|
```

Comment:
Yes, that does make sense to me. And much better than in a serialiser. I think that serialisers shouldn't have any logic except how to *display* things for an API. The only other place could be the view. It could decide to display these payments or not. But since we have only one use case, we don't need to include data we don't need. And data selection / structuring sounds like the kind of thing a model should do.